payment-service/src/main/java/com/example/paymentservice/config/logging/LoggingAspect.java

```diff
@@ -63,22 +71,82 @@ public void logAfterThrowing(JoinPoint joinPoint, Throwable e) {
 
     @Around("applicationPackagePointcut()")
     public Object logAround(ProceedingJoinPoint joinPoint) throws Throwable {
-        if (log.isDebugEnabled()) {
-            log.debug(
-                    "Enter: {}.{}()",
-                    joinPoint.getSignature().getDeclaringTypeName(),
-                    joinPoint.getSignature().getName());
-        }
+        LogLevel logLevel = determineLogLevel(joinPoint);
+        String methodName = joinPoint.getSignature().getName();
+        Class<?> originClass = joinPoint.getTarget().getClass();
+
+        LogWriter.write(originClass, LogLevel.INFO, methodName + "() start execution");
+        logMethodParamsIfEnabled(joinPoint, logLevel, methodName);
+
         long start = System.currentTimeMillis();
         Object result = joinPoint.proceed();
         long end = System.currentTimeMillis();
-        if (log.isDebugEnabled()) {
-            log.debug(
-                    "Exit: {}.{}(). Time taken: {} millis",
-                    joinPoint.getSignature().getDeclaringTypeName(),
-                    joinPoint.getSignature().getName(),
-                    end - start);
-        }
+
+        logMethodResultIfEnabled(joinPoint, result, originClass, logLevel, methodName);
+        LogWriter.write(
+                originClass,
+                LogLevel.INFO,
+                methodName
+                        + "() finished execution and took ("
+                        + (end - start)
+                        + ") mills to execute");
+
         return result;
     }
+
+    private LogLevel determineLogLevel(ProceedingJoinPoint joinPoint) {
+        MethodSignature methodSignature = (MethodSignature) joinPoint.getSignature();
+        Method method = methodSignature.getMethod();
+        Loggable methodAnnotation = method.getAnnotation(Loggable.class);
+        Loggable classAnnotation = joinPoint.getTarget().getClass().getAnnotation(Loggable.class);
+        return Optional.ofNullable(methodAnnotation).orElse(classAnnotation).value();
+    }
+
+    private void logMethodParamsIfEnabled(
+            ProceedingJoinPoint joinPoint, LogLevel logLevel, String methodName) {
+        Loggable methodAnnotation =
```

Comment:
@coderabbitai, Line number 98 to 102, 107 to 113 and 139 to 145 looks similar, refactor this code